### PR TITLE
Docker compose

### PR DESCRIPTION
**Pull request by silverdaz**
Docker Containers started by `docker-compose up -d`

Create the images beforehand with: `make -C images`

Alternative:
```
make -C images gpg2.1   # probably required to first complete
make -C images -j 4     # Targets run in parallel
```

### CODE REVIEW

**Comment by silverdaz** [comment]
Ok, it now works. I can encrypt and decrypt in the worker-container!
No GPG-forwarding yet, I'll do that after the holidays.

File: docker/README.md [hunk 1]
```diff
@@ -6,41 +6,41 @@ It is necessary to first create a `.env` file with the following variables:
 (mostly used to parameterize the docker-compose file itself)
 
 	COMPOSE_PROJECT_NAME=ega
-	CODE=..
+	CODE=..                   # where setup.py is
```
Comment (nanjiangshu) [inline docker/README.md]: Whether relative path for absolute path?

File: docker/README.md [hunk 2]
```diff
@@ -6,41 +6,41 @@ It is necessary to first create a `.env` file with the following variables:
...
 	CONF=<your-conf.ini-file>
```
Comment (nanjiangshu) [inline docker/README.md]: it seems `CONF` should be the path containing the `ini` file

File: docker/README.md (hunk 1 above)
Comment (silverdaz) [inline docker/README.md]: As you wish.
The relative path `..` worked for me.

File: docker/README.md (hunk 2 above)
Comment (silverdaz) [inline docker/README.md]: That's correct. 
I'll update the text to be more clear.

File: docker/README.md [hunk 3]
```diff
@@ -6,41 +6,41 @@ It is necessary to first create a `.env` file with the following variables:
...
-	RSAHOME=<its-location>
-	
+	INBOX=<folder>            # mapped to /home on the ega-inbox
+	STAGING=<folder>          # mapped to /ega/staging on the ega-workers
+	VAULT=<folder>            # mapped to /ega/vault on the ega-vault and ega-verify>
+	RSA_HOME=<folder>         # mapped to /root/.rsa on the ega-workers
+	GPG_HOME=<folder>         # mapped to /root/.gnupg on the ega-gpg-agent
+	GPG_PUBRING=<gpg-pubring> # mapped to /root/.gnupg/pubring.kbx on the ega-workers
+	GPG_PRIV_KEYS=<gpg-keys>  # mapped to /root/.gnupg/private-keys-v1.d on the ega-workers
+	SSH_PUB_KEY=<ssh-key.pub> # mapped to /root/.ssh/ega.pub
```
Comment (nanjiangshu) [inline docker/README.md]: These keys should be the files containing the keys, not the content of the keys, so `GPG_PRIV_KEYS=<gpg-private-key-file>` would be more precise.

File: docker/README.md [hunk 4]
```diff
@@ -6,41 +6,41 @@ It is necessary to first create a `.env` file with the following variables:
...
-SSH_KEY=<your-key-to-login-onto-the-sftp-server>
-```
 
 ## Running
 
 	docker-compose up -d
 	
-If the images are not created, they will be. It takes some time.
+If the images are not created, they will be. <br/>
+However, it is advised to [create them](images) beforehand.
```
Comment (nanjiangshu) [inline docker/README.md]: Potential problems are not addressed, e.g. how to solve the problem when not all containers are UP and RUNNING after 

` docker-compose up -d`

e.g., in my case. the result is like this

```
$ docker-compose ps
        Name                 Command                 State                  Ports         
-----------------------------------------------------------------------------------------
ega-connectors         connectors.sh          Exit 1                                      
ega-db                 docker-entrypoint.sh   Up                     0.0.0.0:5432->5432/t 
                       postgres                                      cp                   
ega-frontend           frontend.sh            Exit 1                                      
ega-inbox              inbox.sh               Exit 1                                      
ega-monitors           monitors.sh            Exit 1                                      
ega-mq                 docker-entrypoint.sh   Up                     15671/tcp, 0.0.0.0:1 
                       rabbi ...                                     5672->15672/tcp,     
                                                                     25672/tcp, 4369/tcp, 
                                                                     5671/tcp, 0.0.0.0:56 
                                                                     72->5672/tcp         
ega-vault              vault.sh               Exit 1                                      
ega-verify             verify.sh              Exit 1                                      
ega-worker-1           worker.sh              Exit 1                                      
ega-worker-2           worker.sh              Exit 1  
```

File: docker/docker-compose.yml [hunk 5]
```diff
@@ -6,33 +6,33 @@ services:
   # The settings in the docker container define
   # the vhosts, users, permissions, queues, exchanges and bindings
   mq:
-    #env_file: ./details/mq.env
-    build: ./mq
+    #env_file: .env.d/mq
```
Comment (nanjiangshu) [inline docker/docker-compose.yml]: the env file for `mq` is not used?

File: docker/entrypoints/connectors.sh [hunk 6]
```diff
@@ -4,12 +4,15 @@ set -e
 
 pip install -e /root/ega
 
-sleep 6
+echo "Waiting for Message Broker"
+until nc -4 --send-only ega-mq 5672 </dev/null &>/dev/null; do sleep 1; done
+echo "Waiting for database"
+until nc -4 --send-only ega-db 5432 </dev/null &>/dev/null; do sleep 1; done
```
Comment (nanjiangshu) [inline docker/entrypoints/connectors.sh]: Is the image based on CentOS?

File: docker/docker-compose.yml (hunk 5 above)
Comment (silverdaz) [inline docker/docker-compose.yml]: No, I baked it them in the image in a secure way. That's ok for the moment

File: docker/entrypoints/connectors.sh (hunk 6 above)
Comment (silverdaz) [inline docker/entrypoints/connectors.sh]: https://github.com/NBISweden/LocalEGA/blob/docker-compose/docker/images/common/Dockerfile#L1

File: docker/README.md (hunk 4 above)
Comment (silverdaz) [inline docker/README.md]: docker-compose up is not concerned about such errors.
Another script could check for them.

Comment (silverdaz) [inline docker/README.md]: Feel free to write one ;)

**Pull request merged**